### PR TITLE
DLSV2-510 Fix to PRN display in Self Assessment Excel Export

### PR DESCRIPTION
**Pull request by Prasanna-Nadella**
### Jira Link
https://hee-dls.atlassian.net/browse/DLSV2-510

### Description
Fix to PRN display in the Excel export to only indicate whether PRN is recorded for supervisors.

### Screenshots
_Attach screenshots on mobile, tablet and desktop._

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Data/Services/CandidateAssessmentDownloadFileService.cs [hunk 1]
```diff
@@ -70,7 +70,7 @@ private static void AddSummarySheet(IXLWorkbook workbook, CandidateAssessmentExp
             rowNum++;
             sheet.Cell(rowNum, 1).Value ="Learner PRN";
             sheet.Cell(rowNum, 1).Style.Fill.BackgroundColor = XLColor.LightBlue;
-            sheet.Cell(rowNum, 2).Value = candidateAssessmentExportSummary.CandidatePrn;
+            sheet.Cell(rowNum, 2).Value = (!string.IsNullOrEmpty(candidateAssessmentExportSummary.CandidatePrn) ? "Recorded" : "Not Recorded");
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Data/Services/CandidateAssessmentDownloadFileService.cs:73]: We want to show the candidates PRN if present, so this line should be something like:
`sheet.Cell(rowNum, 2).Value = (!string.IsNullOrEmpty(candidateAssessmentExportSummary.CandidatePrn) ? candidateAssessmentExportSummary.CandidatePrn.ToString() : "Not Recorded");`

**Review by kevwhitt-hee — COMMENTED** [review]
Looks good. One slight tweak required - see code comments.

**Review by kevwhitt-hee — CHANGES_REQUESTED** [review]
Looks good. One slight tweak required - see code comments.

**Comment by Prasanna-Nadella** [comment]
But the issue says differently. Below is the exact wording from DLSV2-510. Please confirm if you still would like to replace "Recorded" with the PRN.


In both places, we should replace this with “Recorded” if a PRN exists and “Not Recorded” if it is blank.


Thanks & Regards
________________________________

**Comment by kevwhitt-hee** [comment]
The "both places" comment relates to both places where supervisor PRN is currently displayed. Because this is the learner PRN, we should display the actual PRN for the learner if it exists but for the sake of consistency, we should show "Not recorded" if it doesn't.

**Comment by Prasanna-Nadella** [comment]
Hi Kevin,

Good morning.

That makes sense and it's done now.

Checked-in and pushed the changes as well.

Please check it and let me know if anything else is required?

Best Regarads

________________________________

**Review by kevwhitt-hee — CHANGES_REQUESTED** [review]
Noticed a small issue in the details tab. When a result hasn't been confirmed, we still show "Not recorded" in the PRN column, even though there is no reviewer in the reviewer column:
image
We should leave the field blank if the result hasn't been confirmed (i.e. if the Reviewer column is blank).

**Comment by Prasanna-Nadella** [comment]
Amended it by adding Reviewer check to the query.

Please check the attached O/P file and let me know.

Thanks

________________________________

**Review by kevwhitt-hee — APPROVED** [review]
All good.

**Pull request merged**